File: kotlin/scrabble-score/src/main/kotlin/ScrabbleScore.kt
package ScrabbleScore

val letterScore = mapOf("aeioulnrst" to 1,
                        "dg" to 2,
                        "bcmp" to 3,
                        "fhvwy" to 4,
                        "k" to 5,
                        "jx" to 8,
                        "qz" to 10
                        )

fun scoreWord(word: String): Int =
    word.map({ c -> letterLookup(c) }).sum()

fun letterLookup(letter: Char): Int {
    for ((k, v) in letterScore) {
        if (k.contains(letter, true)) {
            return v
        }
    }

    return 0
}

File: python/scrabble-score/scrabble_score.py
STRING_TO_SCORES = {
    "AEIOULNRST": 1,
    "DG": 2,
    "BCMP": 3,
    "FHVWY": 4,
    "K": 5,
    "JX": 8,
    "QZ": 10,
}

LETTER_VALUES = {
    letter: score
    for (string, score) in STRING_TO_SCORES.items()
    for letter in string
}


def score(word):
    return sum(LETTER_VALUES[character] for character in word.upper())
